### ISSUE TITLE
SingleArgumentDig false positive (*)

### ISSUE THREAD

**Issue by johnnyshields** [rubocop/rubocop#12284]
Using Rubocop 1.57.1 with Ruby 3.2 -- latest release of both at time of writing.

```ruby
    # incorrectly triggers SingleArgumentDig
    def foo(*)
      bar.dig(*)
    end
```

This was incorrectly autocorrected to:

```ruby
    def foo(*)
      bar[*]
    end
```

Possibly PR https://github.com/rubocop/rubocop/pull/10576 fixed `...` case but not `*` case, I'm not sure.

**Comment by koic** [comment]
`...` accepts all arguments for delegation, whereas `*` is an anonymized `*args`. Thus, it seems like it should be treated the same as the good case `{ key1: { key2: 'value' } }.dig(*keys)`:
https://docs.rubocop.org/rubocop/1.57/cops_style.html#stylesingleargumentdig

What issues were encountered?

**Comment by johnnyshields** [comment]
@koic when I used autocorrect it corrected it to:

```
     bar[*]
```

which is non-sense. I've updated the ticket description above. Thank you for your fast response as always 🙇‍♂️

**Comment by koic** [comment]
Ah, I understand. This is clearly a bug. Thank you!
